src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -128,9 +128,13 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
 
                             var result = await _fhirService.UpdateResourceAsync(newObservation).ConfigureAwait(false);
 
-                            if (result.VersionId != "1")
+                            // If the version id is not equal to "1" on a create then it is likely two processes modified
+                            // the resource at the same time.
+                            // However we cannot check for not equal to "1" because a resource can be soft-deleted and recreated
+                            // and on the recreate the resource version will also be not equal to "1",
+                            // so instead we will check for resource version id = "2"
+                            if (result.VersionId == "2")

```

Comment:
This is useful but I just want to point out there is nothing in the FHIR spec that versions are incremental integers.  This aligns with how the Microsoft FHIR service works but in scenarios that target other FHIR service implementations (or for example Blobs/Data Lakes) we shouldn't have the same expectation.

Reply:
That is a good callout. While working with Blob it appears that the ETag/Version returned is something like: `"0x8DB2C9BA0A81068"`. It can probably be turned into an integer, but not sure if its incremental. But we can always update this strategy at a later date if we need to support Blob.